shared/protos/garbage/garbage.proto

```diff
@@ -0,0 +1,234 @@
+        {symbol: "TypicalGarbage"  pattern: "projects/{project_id}/typical_garbage/{garbage_id}"},
+        {symbol: "SpecificGarbage" pattern: "projects/{project_id}/specific_garbage/{garbage_id}"},
+        {symbol: "RepeatedGarbage" pattern: "projects/{project_id}/repeated_garbage/{garbage_id}"}
+      ]
+    }];
+  repeated GarbageItem items = 2;
+}
+
+message LongRunningGarbageMetadata {
+  // Approximate percentage of garbage processed thus far. Guaranteed to be 100
+  // when the garbage is fully processed and the items are available.
+  int32 progress_percent = 1;
+
+  // Time when the request was received.
+  google.protobuf.Timestamp start_time = 2;
+
+  // Time of the most recent processing update.
+  google.protobuf.Timestamp last_update_time = 3;
+}
+
+message ListGarbageRequest {
+  string garbage = 1 [
+    (google.api.resource_set) = {
+      symbol: "Garbage"
+      resources: [
+        {symbol: "SimpleGarbage"   pattern: "projects/{project_id}/simple_garbage/{garbage_id}"},
+        {symbol: "TypicalGarbage"  pattern: "projects/{project_id}/typical_garbage/{garbage_id}"},
+        {symbol: "SpecificGarbage" pattern: "projects/{project_id}/specific_garbage/{garbage_id}"},
+        {symbol: "RepeatedGarbage" pattern: "projects/{project_id}/repeated_garbage/{garbage_id}"}
+      ]
+    }];
+}
+
+message ListGarbageResponse {
+  string garbage = 1 [
+    (google.api.resource_set) = {
+      symbol: "Garbage"
+      resources: [
+        {symbol: "SimpleGarbage"   pattern: "projects/{project_id}/simple_garbage/{garbage_id}"},
+        {symbol: "TypicalGarbage"  pattern: "projects/{project_id}/typical_garbage/{garbage_id}"},
+        {symbol: "SpecificGarbage" pattern: "projects/{project_id}/specific_garbage/{garbage_id}"},
+        {symbol: "RepeatedGarbage" pattern: "projects/{project_id}/repeated_garbage/{garbage_id}"}
+      ]
+    }];
+  repeated GarbageItem items = 2;
+}
+
+message GarbageItem {
+  string garbage = 1 [
+    (google.api.resource_set) = {
+      symbol: "Garbage"
+      resources: [
+        {symbol: "SimpleGarbage"   pattern: "projects/{project_id}/simple_garbage/{garbage_id}"},
+        {symbol: "TypicalGarbage"  pattern: "projects/{project_id}/typical_garbage/{garbage_id}"},
+        {symbol: "SpecificGarbage" pattern: "projects/{project_id}/specific_garbage/{garbage_id}"},
+        {symbol: "RepeatedGarbage" pattern: "projects/{project_id}/repeated_garbage/{garbage_id}"}
+      ]
+    }];
+  oneof item {
+    string a = 2;
+    int32  b = 3;
+  }
+}
+
+// A simple garbage resource
+message SimpleGarbage {
+  string name = 1 [(google.api.resource) = {
+    pattern: "projects/{project}/simple_garbage/{garbage}"
+  }];
+}
+
+// A simple garbage item resource
+message SimpleGarbageItem {
+  string garbage = 1 [
+    (google.api.resource_reference) = "SimpleGarbage"];
+  oneof item {
+    string a = 2;
+    int32  b = 3;
+  }
+}
+
+message TypicalGarbage {
```

Comment:
Maybe add a nested type or two for completeness?

Reply:
How do you mean? Isn't `GarbageMap` a nested type?

Reply:
No, I mean something like:

```proto
message Foo {
    message Bar {
        // ....
    }
    // ....
}
```

It's not something we need to do right now, but at some point we will want to make sure the generator can handle nested types correctly.

Reply:
Oooohhh, I didn't even realize we could do that...

Reply:
Added.